### PR TITLE
add auditing events search api

### PR DESCRIPTION
**Pull request by wanjunlei**
What type of PR is this?

/kind feature

What this PR does / why we need it:

Add auditing events search api

### CODE REVIEW

File: pkg/simple/client/auditing/elasticsearch/elasticsearch.go [hunk 1]
```diff
@@ -0,0 +1,391 @@
...
+						"match_phrase": map[string]string{"ObjectRef.Namespace.keyword": k},
+					}, {
+						"range": map[string]interface{}{
+							"RequestReceivedTimestamp": map[string]interface{}{
+								"gte": v,
+							},
+						},
+					}},
+				},
+			})
```
Comment (junotx) [inline pkg/simple/client/auditing/elasticsearch/elasticsearch.go:266]: The case where namespace is empty requires special handling, and you can refer to [this pr](https://github.com/kubesphere/kubesphere/pull/2126) @wanjunlei

Comment (wanjunlei) [reply pkg/simple/client/auditing/elasticsearch/elasticsearch.go:266]: When namespace is empty, the value in es is "", so it dose not need special handling

Comment (junotx) [reply pkg/simple/client/auditing/elasticsearch/elasticsearch.go:266]: So are those fields with empty value also stored in es audit indices ?

Comment (wanjunlei) [reply pkg/simple/client/auditing/elasticsearch/elasticsearch.go:266]: yes

Comment (benjaminhuo) [reply pkg/simple/client/auditing/elasticsearch/elasticsearch.go:266]: So there is no need to adjust this ? @junotx @wanjunlei

Comment (wanjunlei) [reply pkg/simple/client/auditing/elasticsearch/elasticsearch.go:266]: yes

**Comment by benjaminhuo** [comment]
@zryfish is this PR ok for you?

**Pull request merged**